content/blog/2018-04-22-essurvey.md

```diff
@@ -0,0 +1,337 @@
+
+    spain <-
+      import_country(
+        country = "Spain",
+        rounds = 1:7
+        )
+
+`spain` will now be a list of `length(rounds)` containing a data frame
+for each round. The `import_*` family is concerned with downloading the
+data and thus always returns a list containing data frames unless only
+one round is specified, in which it returns a `tibble`. Conversely, the
+`show_*` family grabs information from the ESS website and always
+returns vectors.
+
+To download all rounds for a country automatically you can use
+`import_all_cntrounds`.
+
+    import_all_cntrounds("Spain")
+
+Getting back to our example, ESS related datasets have missing
+categories coded differently between questions. For example, questions
+with scales from `1:5`, have missing categories such as "Don't know" and
+"Refusal" coded between `7` and `10`. But questions with scales from
+`1:10` are coded as `77`, `88` and `99`. `recode_missings` accepts a
+`tibble` as a main argument and automatically returns a new `tibble`
+with all missing values recoded as `NA`. You should checkout
+`?recode_missings` for more details and elaborated examples.
+
+**Note**: I urge the reader not to recode these categories to missing
+without previously investigating the importance of these categories.
+
+For example, let's recode missing values in all Spanish waves, bind them
+into one single `tibble` and visualize how satisfied are Spaniards with
+their government. First, let's extract a cleaner data.
+
+    semi_cleaned <-
+      spain %>% 
+      map(recode_missings) %>% 
+      bind_rows() %>% 
+      mutate(name = str_sub(name, end = 4)) %>% 
+      select(name, stfgov)
+
+    semi_cleaned
+
+    ## # A tibble: 13,543 x 2
+    ##    name  stfgov
+    ##    <chr>  <dbl>
+    ##  1 ESS1      0.
+    ##  2 ESS1      0.
+    ##  3 ESS1      5.
+    ##  4 ESS1      5.
+    ##  5 ESS1      4.
+    ##  6 ESS1      7.
+    ##  7 ESS1      2.
+    ##  8 ESS1      2.
+    ##  9 ESS1      3.
+    ## 10 ESS1      3.
+    ## # ... with 13,533 more rows
+
+There we go. The scale of `stfgov` is between `0` and `10`, let's
+collapse that into smaller categories, calculate the percentage of
+respondents within each category and visualize the change over time.
+
+    semi_cleaned %>% 
+      mutate(stfgov = case_when(stfgov <= 3 ~ "Low",
+                                between(stfgov, 4, 6) ~ "Mid",
+                                stfgov >= 7 ~ "High"),
+             stfgov = factor(stfgov, levels = c("Low", "Mid", "High"))) %>%
+      count(name, stfgov) %>% 
+      group_by(name) %>% 
+      mutate(perc = n / sum(n)) %>% 
+      ggplot(aes(name, perc, group = stfgov, colour = stfgov)) +
+      geom_line() +
+      theme_bw() +
+      labs(x = "ESS rounds", y = "Satisfaction with Government (%)") +
+      scale_colour_discrete(name = NULL)
+
+![](/img/blog-images/2018-04-22-essurvey/spain_stf-1.png)
+
+Looks like Spaniards are very unhappy with their government!

```

Comment:
If I understand correctly, ESS1 to ESS7 are data collected from surveys every two years (can't recall where I inferred this). Is it possible (in the text for ex) to indicate the years or time span? Can you say "Spaniards are increasingly unhappy with their government!"?